helpers/crearMovimientoDetalle.go

```diff
@@ -0,0 +1,311 @@
+	var err error
+
+	if idMovProcExternoCast, err = strconv.Atoi(idMovProcExterno); err != nil {
+		logs.Error(err)
+		outputError := errorctrl.Error("RegistroMovimientoDetalle - strconv.Atoi(idMovProcExterno)", err, "400")
+		return models.MovimientoDetalle{}, outputError
+	}
+
+	// logs.Debug("INSERTAR movimiento: ", idMovProcExternoCast)
+
+	registroMovProcExterno = models.MovimientoProcesoExterno{
+		Id: idMovProcExternoCast,
+	}
+
+	var nuevoDeltaAcum float64
+	var nuevoSaldo float64
+	var nuevoValor float64
+	var err2 map[string]interface{}
+	var nuevoDetalleCuenPre map[string]interface{}
+
+	if nuevoDeltaAcum, nuevoSaldo, nuevoValor, err2 = CalcularMontos(detalleCuenPre, saldo, valor, publicar); err2 != nil {
+		logs.Error(err2)
+		outputError := errorctrl.Error("RegistroMovimientoDetalle - CalcularDeltaAcum(detalleCuenPre)", err2, "500")
+		return models.MovimientoDetalle{}, outputError
+	}
+
+	if err := json.Unmarshal([]byte(detalleCuenPre), &nuevoDetalleCuenPre); err != nil {
+		logs.Error(err)
+		outputError := errorctrl.Error("RegistroMovimientoDetalle - json.Unmarshal([]byte(detalleCuenPre), &nuevoDetalleCuenPre)", err, "400")
+		return models.MovimientoDetalle{}, outputError
+	}
+
+	deltaEnviado := map[string]interface{}{
+		"DeltaAcum": nuevoDeltaAcum,
+	}
+
+	if err := mergo.Merge(&nuevoDetalleCuenPre, deltaEnviado); err != nil {
+		logs.Error(err)
+		outputError := errorctrl.Error("RegistroMovimientoDetalle - mergo.Merge(&new_detalle_CuenPre, deltaEnviado)", err, "400")
+		return models.MovimientoDetalle{}, outputError
+	}
+
+	for k, detalle := range nuevoDetalleCuenPre {
+		if k != "RubroId" {
+			switch detalle.(type) {
+			case string:
+				if nuevoDetalleCuenPre[k], err = strconv.Atoi(detalle.(string)); err != nil {
+					logs.Error(err)
+					outputError := errorctrl.Error("RegistroMovimientoDetalle - strconv.Atoi(detalle.(string))", err, "400")
+					return models.MovimientoDetalle{}, outputError
+				}
+			default:
+				logs.Warn("La variable no es de tipo string")
+			}
+
+		}
+	}
+
+	var nuevoDetalleCuenPreCast []byte
+
+	if nuevoDetalleCuenPreCast, err = json.Marshal(nuevoDetalleCuenPre); err != nil {
+		logs.Error(err)
+		outputError := errorctrl.Error("RegistroMovimientoDetalle - json.Marshal(nuevoDetalleCuenPre)", err, "400")
+		return models.MovimientoDetalle{}, outputError
+	}
+
+	registroMovimientoDetalleRespuesta = models.MovimientoDetalle{
+		Activo:                     true,
+		Descripcion:                "Creación de movimiento detalle",
+		Detalle:                    string(nuevoDetalleCuenPreCast),
+		MovimientoProcesoExternoId: &registroMovProcExterno,
+		Saldo:                      nuevoSaldo,
+		Valor:                      nuevoValor,
+	}
+
+	return registroMovimientoDetalleRespuesta, nil
+}
+
+func CalcularMontos(detalleCuenPre string, saldo float64, valor float64, publicar bool) (detalAcumRespuesta float64, saldoRespuesta float64, valorRespuesta float64, outputError map[string]interface{}) {
```

Comment:
1. Linea muy larga
2. Falta documentar, cambiar el `...` del _suggested change_
```suggestion
// CalcularMontos ...
func CalcularMontos(detalleCuenPre string,
saldo float64, valor float64, publicar bool) (
detalAcumRespuesta float64, saldoRespuesta float64, valorRespuesta float64,
outputError map[string]interface{}) {
```